app/services/chat_channels/create_with_users.rb

```diff
@@ -0,0 +1,43 @@
+module ChatChannels
+  class CreateWithUsers
+    def initialize(users: [], channel_type: "direct", contrived_name: "New Channel", membership_role: "member")
+      @users = users
+      @channel_type = channel_type
+      @contrived_name = contrived_name
+      @membership_role = membership_role
+    end
+
+    attr_reader :users, :channel_type, :membership_role
+    attr_accessor :contrived_name
```

Comment:
are these used outside? otherwise they can be set to `private`. also I don't think you need `contrived_name` to be an accessor. 

I would use `attr_reader :users, :channel_type, :contrived_name, :membership_role`